gradoop-flink/src/main/java/org/gradoop/flink/model/api/epgm/BaseGraphOperators.java

```diff
@@ -0,0 +1,247 @@
+  G extends EPGMGraphHead,
+  V extends EPGMVertex,
+  E extends EPGMEdge,
+  LG extends BaseGraph<G, V, E, LG>> {
+
+  //----------------------------------------------------------------------------
+  // Unary Operators
+  //----------------------------------------------------------------------------
+
+  /**
+   * Creates a copy of the logical graph.
+   * <p>
+   * Note that this method creates new graph head, vertex and edge instances.
+   *
+   * @return projected logical graph
+   */
+  LG copy();
+
+  /**
+   * Transforms the elements of the logical graph using the given transformation functions.
+   * The identity of the elements is preserved.
+   *
+   * @param graphHeadTransformationFunction graph head transformation function
+   * @param vertexTransformationFunction    vertex transformation function
+   * @param edgeTransformationFunction      edge transformation function
+   * @return transformed logical graph
+   */
+  LG transform(
+    TransformationFunction<G> graphHeadTransformationFunction,
+    TransformationFunction<V> vertexTransformationFunction,
+    TransformationFunction<E> edgeTransformationFunction);
+
+  /**
+   * Transforms the graph head of the logical graph using the given
+   * transformation function. The identity of the graph is preserved.
+   *
+   * @param graphHeadTransformationFunction graph head transformation function
+   * @return transformed logical graph
+   */
+  LG transformGraphHead(
+    TransformationFunction<G> graphHeadTransformationFunction);
+
+  /**
+   * Transforms the vertices of the logical graph using the given transformation
+   * function. The identity of the vertices is preserved.
+   *
+   * @param vertexTransformationFunction vertex transformation function
+   * @return transformed logical graph
+   */
+  LG transformVertices(TransformationFunction<V> vertexTransformationFunction);
+
+  /**
+   * Transforms the edges of the logical graph using the given transformation function.
+   * The identity of the edges is preserved.
+   *
+   * @param edgeTransformationFunction edge transformation function
+   * @return transformed logical graph
+   */
+  LG transformEdges(TransformationFunction<E> edgeTransformationFunction);
+
+  /**
+   * Returns the subgraph that is induced by the vertices which fulfill the given filter function.
+   *
+   * @param vertexFilterFunction vertex filter function
+   * @return vertex-induced subgraph as a new logical graph
+   */
+  LG vertexInducedSubgraph(FilterFunction<V> vertexFilterFunction);
+
+  /**
+   * Returns the subgraph that is induced by the edges which fulfill the given filter function.
+   *
+   * @param edgeFilterFunction edge filter function
+   * @return edge-induced subgraph as a new logical graph
+   */
+  LG edgeInducedSubgraph(FilterFunction<E> edgeFilterFunction);
+
+  /**
+   * Returns a subgraph of the logical graph which contains only those vertices
+   * and edges that fulfil the given vertex and edge filter function respectively.
```

Comment:
This sentence kind of makes it sound like there is only one filter function, consider rephrasing it.